src/util/ParseableDuration.h

```diff
@@ -39,26 +65,24 @@ class ParseableDuration {
       };
 
       if (unit == "ns") {
-        result = toDuration.template operator()<nanoseconds>(amount);
+        return toDuration.template operator()<nanoseconds>(amount);
       } else if (unit == "us") {
-        result = toDuration.template operator()<microseconds>(amount);
+        return toDuration.template operator()<microseconds>(amount);
       } else if (unit == "ms") {
-        result = toDuration.template operator()<milliseconds>(amount);
+        return toDuration.template operator()<milliseconds>(amount);
       } else if (unit == "s") {
-        result = toDuration.template operator()<seconds>(amount);
+        return toDuration.template operator()<seconds>(amount);
       } else if (unit == "min") {
-        result = toDuration.template operator()<minutes>(amount);
+        return toDuration.template operator()<minutes>(amount);
       } else if (unit == "h") {
-        result = toDuration.template operator()<hours>(amount);
+        return toDuration.template operator()<hours>(amount);
```

Comment:
The pattern `} else { AD_CORRECTNESS_CHECK(unit == "h"); return ....}`
will make the coverage tool happier, and also indicates more the exhaustiveness (it really shouldn't happen because of the regex.